### PR TITLE
feat(flamegraph): fix recursion coloring

### PR DESCRIPTION
**Pull request by JonasBa**
Recursion requires unique frame keys, because we were using just frame names, some frames were marked as recursive in parts of the stack where there was no recursions

### CODE REVIEW

File: static/app/utils/profiling/colors/utils.tsx [hunk 1]
```diff
@@ -114,11 +111,11 @@ export function toRGBAString(r: number, g: number, b: number, alpha: number): st
   )}, ${alpha})`;
 }
 
-export function defaultFrameSortKey(frame: Frame): string {
-  return (frame.file || '') + frame.name;
+export function defaultFrameSortKey(frame: FlamegraphFrame): string {
+  return frame.frame.name + (frame.frame.file || '');
```
Comment (Zylphrex) [inline static/app/utils/profiling/colors/utils.tsx:115]: Why did name move before file?

File: static/app/components/profiling/FlamegraphZoomView.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import * as React from 'react';
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
```
Comment (Zylphrex) [inline static/app/components/profiling/FlamegraphZoomView.tsx:1]: Should we remove the default import for `React` completely while we're here?

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/FlamegraphZoomView.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/profiling/FlamegraphZoomView.tsx:1]: Nah, we use a fragment below in the render code, that's why I kept it. Kinda annoying :/

File: static/app/utils/profiling/colors/utils.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/utils/profiling/colors/utils.tsx:115]: We always have a name, but we dont always have the file/binary, so I wanted to have that out first

**Pull request merged**